src/provider.ts

```diff
@@ -233,6 +259,18 @@ export class Provider {
                     if (response) {
                         clearInterval(intervalId);
                         this.requests.delete(requestId);
+                        if (isUnauthMessage(response)) {
+                            const code = extractUnauthCode(response);
+                            this.hooks?.onRequestFinish?.({
```

Comment:
If we already do `onRequestFinish` , with status: 'error' maybe we don't need the `onUnauth` hook and isntead we need to document that user need to call onRequestFinish and implement that instead?

Again, less thing to manage for us